src/cargo/sources/registry/mod.rs

```diff
@@ -369,9 +369,65 @@ impl<'a> RegistryDependency<'a> {
     }
 }
 
+/// An indicator that the prefetching for a given package has completed.
+///
+/// To retrieve the index data for the package, use `Summaries::parse`.
+pub struct Fetched {
+    name: InternedString,
+    path: PathBuf,
+    // NOTE: with https://github.com/steveklabnik/semver/issues/170 the HashSet is unnecessary
```

Comment:
I think we want union not intersection for this. If we have a crate that requests `^2.0` and one that does `^3.0` then the intersection is the empty set, but we will be picking one from each. So we need to prefetch them all.

Even if we have a summary that needs `>=2.3, <3.0` and a different summary that needs `>2.0, <=2.7`. We can't combine them into `>=2.3, <=2.7` as it is possible that the resolver will only activate one or the other of the summaries.

Reply:
Yup, you're exactly right, I had that thought right after our conversation as well :sweat_smile: